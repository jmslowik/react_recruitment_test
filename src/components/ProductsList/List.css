.container{
    width: 1000px;
    margin: 100px auto 0 auto;
}

.list {
    margin-bottom: 0;
    border-bottom-style: solid;
}

.list .row:last-child {
    border-bottom-style: none;
}

.sum {
    font-weight: bold;
}