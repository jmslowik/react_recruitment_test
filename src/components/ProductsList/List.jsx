import React from 'react';
import './List.css';

// TODO name as prop

const List = ({ children, sum }) => (
  <div className="container">
    <h3>Lista produkt√≥w</h3>
    <ul className="list">
      { children }
    </ul>
    <ul className="sum">
      { sum }
    </ul>
  </div>
);

export default List;
