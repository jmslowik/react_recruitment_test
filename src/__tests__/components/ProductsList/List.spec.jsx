import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import List from '@/components/ProductsList/List';

describe('List', () => {
  const Child = () => <div className="child">child</div>;
  const Sum = () => <div className="sum-prop">sum</div>;

  it('should render without throwing an error', () => {
    const wrapper = shallow(<List />);
    expect(wrapper.is('.container')).toBeTruthy();
  });

  it('should contain list name', () => {
    const name = 'Lista produktów';
    const wrapper = shallow(<List name={name} />);
    expect(wrapper.contains(<h3>{name}</h3>)).toBeTruthy();
  });

  it('should contain children under the .list', () => {
    const wrapper = mount(
      <List>
        <Child />
        <Child />
      </List>,
    );
    expect(wrapper.find('.list').find('.child')).toHaveLength(2);
  });

  it('should contain sum under the .sum', () => {
    const wrapper = mount(<List sum={<Sum />} />);
    expect(wrapper.find('.sum').find('.sum-prop')).toHaveLength(1);
  });

  it('should render correctly', () => {
    const tree = renderer.create(
      <List name="Lista produktów" sum={<Sum />}>
        <Child />
        <Child />
      </List>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
