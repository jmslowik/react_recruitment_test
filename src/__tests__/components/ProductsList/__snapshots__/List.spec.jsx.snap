// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
<div
  className="container"
>
  <h3>
    Lista produkt√≥w
  </h3>
  <ul
    className="list"
  >
    <div
      className="child"
    >
      child
    </div>
    <div
      className="child"
    >
      child
    </div>
  </ul>
  <ul
    className="sum"
  >
    <div
      className="sum-prop"
    >
      sum
    </div>
  </ul>
</div>
`;
